docs/DeveloperGuide.md

```diff
@@ -1,38 +1,190 @@
+<!-- omit in toc -->
 # Developer Guide
+<!-- omit in toc -->
+## Table of contents
+
+- [General Overview](#general-overview)
+- [Design \& implementation](#design--implementation)
+  - [UI class](#ui-class)
+  - [DataStorage class](#datastorage-class)
+    - [loadChassis() method](#loadchassis-method)
+  - [Parser class](#parser-class)
+  - [PC Class Architecture](#pc-class-architecture)
+  - [Command class](#command-class)
+    - [`list [COMPONENT]` Command (builder mode)](#list-component-command-builder-mode)
+- [Appendix: Requirements](#appendix-requirements)
+  - [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+  - [User Stories](#user-stories)
+  - [Non-Functional Requirements](#non-functional-requirements)
+  - [Glossary](#glossary)
+  - [Instructions for manual testing](#instructions-for-manual-testing)
+
+
+## General Overview
+below is the overall architecture diagram for how BigPP works.
+
+![Architecture Diagram](uml-pictures/ArchitectureDiagram.png)
+
+The program will first load the `UserJson` and files in the `Resources` folder to populate its internal memory of `PCLists` and `ComponentLists`. This will be stored in its `DataStorage`. The `User`'s interaction with the `UI` will be `parsed` into a `command` which would update the `DataStorage` and eventually update the `Menu` which is displayed back to the `User`. This would continue until the `User` exits the program, which would result in the data stored in `DataStorage` being saved into the `UserJson`.
 
-## Acknowledgements
+## Design & implementation
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### UI class
+Below is the Class diagram for the UI class
 
-## Design & implementation
+![UI Class Diagram](uml-pictures/UIClassDiagram.png)
+
+The `UI` Class serves as the interface between the user and BigPP. It is responsible for the following key actions
```

Comment:
In the UI diagram, UI class is declared as an abstract class, but in the explanation, "interface" is used, so it is slightly confusing if UI is an abstract class or an interface